Sources/SwiftFusion/Geometry/Rot2.swift

```diff
@@ -1,121 +1,60 @@
 import TensorFlow
 
-// We need a special version of atan2 that provides a derivative.
-@differentiable
-func atan2wrap(_ s: Double, _ c: Double) -> Double {
-  atan2(s, c)
-}
-
-// Implement derivative of atan2wrap.
-// d atan2(s,c)/s = c / (s^2+c^2)
-// d atan2(s,c)/s = -s / (s^2+c^2)
-// TODO(frank): make use of fact that s^2 + c^2 = 1
-@derivative(of: atan2wrap)
-func _vjpAtan2wrap(_ s: Double, _ c: Double) -> (value: Double, pullback: (Double) -> (Double, Double)) {
-  let theta = atan2(s, c)
-  let normSquared = c * c + s * s
-  return (theta, { v in (v * c / normSquared, -v * s / normSquared) })
-}
-
-// @derivative(of: bar)
-// public func _(_ x: Float) -> (value: Float, differential: (Float) -> Float) {
-//   (x, { dx in dx })
-// }
-
 /// Rot2 class is the Swift type for the SO(2) manifold of 2D Rotations around
 /// the origin.
-public struct Rot2: Equatable, Differentiable, KeyPathIterable {
-  // TODO: think about the situations where need exact value instead of
-  // equivalent classes
-  // var theta_ : Double;
+public struct Rot2: Manifold, Equatable, KeyPathIterable {
 
-  /// Cosine and Sine of the rotation angle
-  private var c_, s_: Double
+  // MARK: - Manifold conformance
 
-  @differentiable
-  public var c: Double {
-    c_
-  }
+  public var coordinateStorage: Rot2Coordinate
+  public init(coordinateStorage: Rot2Coordinate) { self.coordinateStorage = coordinateStorage }
 
-  @usableFromInline
-  @derivative(of: c)
-  func _vjpCos() -> (value: Double, pullback: (Double) -> TangentVector) {
-    (c_, { v in Vector1(-v * self.s_) })
+  public mutating func move(along direction: Coordinate.LocalCoordinate) {
+    coordinateStorage = coordinateStorage.global(direction)
   }
```

Comment:
Could this definition of `move(along:)` be made a default implementation for `protocol Manifold`? It seems to use only protocol requirements.